src/SubtitleOctopus.cpp

```diff
@@ -305,23 +319,27 @@ class SubtitleOctopus {
         ass_set_margins(ass_renderer, top, bottom, left, right);
     }
 
-    int getEventCount() {
+    int getEventCount() const {
         return track->n_events;
     }
 
     int allocEvent() {
+        free(m_is_event_animated);
+        m_is_event_animated = NULL;
```

Comment:
Not sure about `allocEvent`/`removeEvent`. You just reset array `m_is_event_animated`, but there are may be other events, no?
Will `rescanAllAnimations` be called somehow after that?

Reply:
It's a public method now, so if a user of this class decides to muck with events manually he/she assumes full responsibility to rescan them when finished.

I don't see us doing so (mucking with events) anyway, so this is kind of a safety net.